### PR TITLE
Headcrabs

### PR DESCRIPTION
**Pull request by Terilia**
Are you afraid in the dark? Do you only walk in maintenance with a crowbar? Then this PR is for you. 

Headcrabs! 

image
image

The sprites have been done by Ionward! Thank you!

This PR adds a mid-tier random event, which causes, just like in infestation, several headcrabs to spawn in maintenance. (Up to 6)

The headcrabs will attack players on sight, do 8-16 damage and will, if they kill their target, take them over as zombies. The code is very similar (and stolen) from the blob pods. If killed, the headcrab dies and the controlled person can be dragged to medical to be revived. 

On range they will also leap at their enemies every once in a while. 

Additional note: I had to change the name of the headslugs in the file system and in the code. They have been adjusted from simple_animals/hostile/headcrab to simple_animals/hostile/headslug - I have checked to ensure that changeling still works. 

**Changelog:**
:cl:
add: added headcrabs
tweak: adjusted the path's for headslugs. They are not called headcrab anymore
/:cl:

### CODE REVIEW

**Comment by pinatacolada** [comment]
i'd imagine the half life sounds are copyrighted

**Comment by SkeletalElite** [comment]
adjusting the path for headslugs is more of a backend change and doesn't really need to go the changelog, since nothing about headslugs has changed.

**Comment by Bookdrizzt** [comment]
Maybe look into the Romerol Virus PR? Ask the guy who did that if you could look at their code?

**Comment by Ralta** [comment]
Nice idea, I like it. An interesting mid-round antag.

**Comment by Crazylemon64** [comment]
Close/opening to force appveyor update

**Comment by Terilia** [comment]
Is now ready to be merged/reviewed.

**Comment by Terilia** [comment]
Should be ready for review now

**Comment by Ralta** [comment]
Love the new sprites! Great work, @Ionward

**Comment by Couls** [comment]
They seem really weak actually especially since only one spawns in an area but I guess those numbers can be modified after it’s merged

**Comment by Terilia** [comment]
> They seem really weak actually especially since only one spawns in an area but I guess those numbers can be modified after it’s merged
> 
> Though I’d like to see more in one area to give crew a bigger challenge dealing with them

```	switch(headcrab) //Switch is for the future
		if(HEADCRAB_NORMAL)
			spawn_types = list(/mob/living/simple_animal/hostile/headcrab)
			max_number = 6

 	spawn(0)
		var/num = rand(2,max_number)
```

It spawns 2 to 6

**Comment by Couls** [comment]
yeah I read over the code and realized I was wrong my bad! It seems fine now that I’ve gone over it

File: code/modules/events/headcrabs.dm [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		while(turfs.len > 0 && num > 0)
+			var/turf/simulated/floor/T = pick(turfs)
+			turfs.Remove(T)
+			num--
+			var/spawn_type = pick(spawn_types)
+			new spawn_type(T)
+
+
+/datum/event/headcrabs/announce()
+	event_announcement.Announce("Bioscans indicate that creatues have been breeding in [locstring]. Clear them out, before this starts to affect productivity.", "Lifesign Alert")
```
[the remark this answers is not in the document]
Comment (Terilia) [reply code/modules/events/headcrabs.dm]: Crap, good catch

**Comment by Emanthealmighty** [comment]
I think this has a lot of potential!
Many other things could be done to make them more fleshed out... like adding the headcrab lifecycle (baby headcrab, headcrab, gonarch) and/or the other variants featured in half life 2, each having their own strengths and weaknesses.

**Comment by Terilia** [comment]
It does. For now I want it as this prototype that can be spawned as midrounds to see how users react. And if they like it, the idea was from the beginning to flesh them out a little more.

**Comment by Spartan6** [comment]
This should do severe brain damage to the body that medical would need to fix, due to... well, the probe from the headcrab literally inserting itself into the brain of the victim to control its actions.

File: code/modules/events/headcrabs.dm [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+	var/list/spawn_types = list()
+	var/max_number
+	headcrab = 0 //rand(0,x) for the future
+	switch(headcrab) //Switch is for the future
+		if(HEADCRAB_NORMAL)
+			spawn_types = list(/mob/living/simple_animal/hostile/headcrab)
+			max_number = 6
+
+	spawn(0)
```
Comment (Couls) [inline code/modules/events/headcrabs.dm]: Is this necessary?

Comment (Terilia) [reply code/modules/events/headcrabs.dm]: I don't know (I also made a review by accident, woopsie)

**Comment by Terilia** [comment]
Headcrabs can now leap.

**Pull request merged**